src/ReservoirRouter.sol

```diff
@@ -0,0 +1,291 @@
+
+        if (pair == address(0)) {
+            pair = factory.createPair(tokenA, tokenB, curveId);
+        }
+
+        (uint256 reserveA, uint256 reserveB) = ReservoirLibrary.getReserves(address(factory), tokenA, tokenB, curveId);
+        if (reserveA == 0 && reserveB == 0) {
+            (amountA, amountB) = (amountADesired, amountBDesired);
+        }
+        else {
+            uint amountBOptimal = ReservoirLibrary.quote(amountADesired, reserveA, reserveB);
+            if (amountBOptimal <= amountBDesired) {
+                require(amountBOptimal >= amountBMin, "RR: INSUFFICIENT_B_AMOUNT");
+                (amountA, amountB) = (amountADesired, amountBOptimal);
+            } else {
+                uint amountAOptimal = ReservoirLibrary.quote(amountBDesired, reserveB, reserveA);
+                assert(amountAOptimal <= amountADesired);
+                require(amountAOptimal >= amountAMin, "RR: INSUFFICIENT_A_AMOUNT");
+                (amountA, amountB) = (amountAOptimal, amountBDesired);
+            }
+        }
+    }
+
+    function addLiquidity(
+        address tokenA,
+        address tokenB,
+        uint256 curveId,
+        uint amountADesired,
+        uint amountBDesired,
+        uint amountAMin,
+        uint amountBMin,
+        address to
+    ) external payable returns (uint256 amountA, uint256 amountB, uint256 liquidity) {
+        address pair;
+        (amountA, amountB, pair) = _addLiquidity(tokenA, tokenB, curveId, amountADesired, amountBDesired, amountAMin, amountBMin);
+
+        _pay(tokenA, msg.sender, pair, amountA);
+        _pay(tokenB, msg.sender, pair, amountB);
+
+        liquidity = IReservoirPair(pair).mint(to);
+    }
+
+    function removeLiquidity(
+        address tokenA,
+        address tokenB,
+        uint256 curveId,
+        uint256 liquidity,
+        uint256 amountAMin,
+        uint256 amountBMin,
+        address to // when withdrawing to the router in UniV3, this address is 0. But we can just put the router's address
+    ) external /*payable*/ returns (uint256 amountA, uint256 amountB) {
+        address pair = ReservoirLibrary.pairFor(address(factory), tokenA, tokenB, curveId);
+        IReservoirPair(pair).transferFrom(msg.sender, pair, liquidity); // send liquidity to pair
+        (uint amount0, uint amount1) = IReservoirPair(pair).burn(to);
+
+        (address token0,) = ReservoirLibrary.sortTokens(tokenA, tokenB);
+        (amountA, amountB) = tokenA == token0 ? (amount0, amount1) : (amount1, amount0);
+
+        require(amountA >= amountAMin, "RR: INSUFFICIENT_A_AMOUNT");
+        require(amountB >= amountBMin, "RR: INSUFFICIENT_B_AMOUNT");
+    }
+
+    // **** SWAP ****
+    // requires the initial amount to have already been sent to the first pair
+    /// @param inOrOut true for exact in, false for exact out
+    function _swap(uint[] memory amounts, bool inOrOut, address[] memory path, uint256[] memory curveIds, address _to) internal {
```

Comment:
We use `_to` to avoid collisions, if we just use `aTo` this problem goes away

Reply:
okay, with ref to the previous comment about style, will change all of them